### DISCUSSION TITLE
Consistency in Results Reporting

### DISCUSSION THREAD

**Discussion by dorin100** [input-output-hk/quality-engineering#27] [General]
As we continue to refine our [Quality Strategy](https://input-output.atlassian.net/wiki/spaces/CI/pages/3789128005/Quality+Strategy+Overview) at the Chapter level, it's become clear that we need to align on how we report test results inside and across all our tribes in a standardized way. To accomplish this, we are inviting everyone to propose solutions that enable us to use similar tools, reuse code and tools whenever possible, and report on all test levels executed inside the tribe.

Standardizing test results reporting at Tribe or even Company level is crucial for several reasons:
- Consistency: Standardizing test results reporting ensures that all teams across the tribe/company are using the same methods, tools, and processes to report test results. This leads to consistency in how the results are interpreted and used to make decisions, helping to avoid confusion or discrepancies in the reporting.
- Efficiency: Standardizing test results reporting helps teams to share tools and code, reduce duplication of efforts, and increase efficiency across the company. This reduces wasted time and resources, allowing teams to focus on other important tasks.
- Transparency: Standardized reporting provides transparency and visibility into the overall quality of the software development process, enabling management and stakeholders to make informed decisions based on reliable data. This facilitates collaboration between teams, makes it easier to track testing progress, and identify areas for improvement.
- Continuous improvement: Standardized test results reporting facilitates the collection of consistent data, which can be analyzed to identify trends and areas for improvement in the testing process. This helps to ensure that testing efforts are continually improving, which in turn leads to better software quality and customer satisfaction.
- Collaboration: Standardizing test results reporting can help promote collaboration among teams by providing a common language and framework for reporting test results. This can help improve communication and understanding between teams, and promote a culture of knowledge-sharing and collaboration.

The standardization of the test results can be done in multiple steps:
1. reporting of the test results at the Tribe/Product level
2. alignment on tools and practices: we can analyze if there are any tools or code that can be reused to improve the results reporting across teams and tribes.

[Here is a list of Tools and Technologies](https://input-output.atlassian.net/wiki/spaces/CI/pages/3764420710/Tools+and+Technologies) we already use for the reporting of the system-level tests in our projects.

Our goal is to create consistency in how we report test results across teams and tribes, enabling us to better understand our overall test results and identify areas for improvement. We encourage everyone to contribute their ideas, insights, and suggestions as we work together towards this goal.

Please let us know if you have any questions or concerns, and we look forward to hearing your proposals.

**Comment by sp-io** [comment]
Interesting tool for gathering reports across multiple frameworks : https://reportportal.io/

**Comment by dorin100** [comment]
We started a doc to guide collecting the [test results reporting requirements](https://input-output.atlassian.net/wiki/spaces/CI/pages/3849519107/Test+Results+Reporting+-+Requirements) at the company level.
